compiler/noirc_evaluator/src/errors.rs

```diff
@@ -82,7 +82,7 @@ impl From<SsaReport> for CustomDiagnostic {
                         ("This variable contains a value which is constrained to be a constant. Consider removing this value as additional return values increase proving/verification time".to_string(), call_stack)
                     },
                     InternalWarning::VerifyProof { call_stack } => {
-                        ("verify_proof(...) aggregates data for the verifier, the actual verification will be done when the full proof is verified using nargo verify. nargo prove may generate an invalid proof if bad data is used as input to verify_proof".to_string(), call_stack)
+                        ("The validity of the proof passed to verify_proof(...) can only be checked by the proving system so the ACVM will then defer checking of these proofs to the proving system. Passing an invalid proof is expected to cause the proving system to either fail to generate a proof or generate a proof which fails verification".to_string(), call_stack)
```

Comment:
```suggestion
                        ("The validity of the proof passed to verify_proof(...) can only be checked by the proving backend so the ACVM will then defer checking of these proofs to the proving backend. Passing an invalid proof is expected to cause the proving backend to either fail to generate a proof or generate a proof which fails verification".to_string(), call_stack)
```

Minor nit on wording (to better align with wording in Noir docs)